Source/Laser/LaserProfilesImpl/LaserProfileFromFile.cpp

```diff
@@ -302,14 +322,28 @@ WarpXLaserProfiles::FromFileLaserProfile::read_data_t_chunk(int t_begin, int t_e
         auto E = i.meshes["laserEnvelope"];
         auto E_laser = E[io::RecordComponent::SCALAR];
         openPMD:: Extent full_extent = E_laser.getExtent();
-        openPMD::Extent read_extent = {(i_last - i_first + 1), full_extent[1], full_extent[2]};
-        auto x_data = E_laser.loadChunk< std::complex<double> >(io::Offset{i_first, 0, 0}, read_extent);
-        const int read_size = (i_last - i_first + 1)*m_params.nx*m_params.ny;
-        series.flush();
-        for (int j=0; j<read_size; j++) {
-            h_E_lasy_data[j] = Complex{
-                static_cast<amrex::Real>(x_data.get()[j].real()),
-                static_cast<amrex::Real>(x_data.get()[j].imag())};
+        if (m_params.fileGeom=="thetaMode") {
+            openPMD::Extent read_extent = { full_extent[0], (i_last - i_first + 1), full_extent[2]};
+            auto r_data = E_laser.loadChunk< std::complex<double> >(io::Offset{ 0, i_first,  0}, read_extent);
+            const int read_size = (i_last - i_first + 1)*m_params.nr;
+            series.flush();
```

Comment:
You could do:
`h_E_lasy_data.resize( read_size*m_params.n_rz_azimuthal_components )`